hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -1012,5 +1114,7 @@ object Simplify {
       val needsDense = sparsityStrategy == NeedsDense || sparsityStrategy.exists(leftBlock = false, rightBlock = true)
       val maybeDense = if (needsDense) BlockMatrixDensify(left) else left
       BlockMatrixMap(maybeDense, leftName, Subst(f, BindingEnv.eval(rightName -> getElement)), needsDense)
+    case BlockMatrixMap(matrix, name, Ref(x, _), _)  if name == x =>
+      matrix
```

Comment:
The new simplify rules could also create a map with a constant body. We could catch those cases with something like
```
  case BlockMatrixMap(matrix, name, const@(Ref(x, _) if name != x | _ if IsConstant(const)), _) =>
    val typ = matrix.typ
    BlockMatrixBroadcast(
      ValueToBlockMatrix(const, FastIndexedSeq(1, 1), typ.blockSize),
      FastIndexedSeq(),
      typ.shape,
      typ.blockSize)
```

Reply:
Oh good idea. Thank you :)